internal/controller/nats/integrationtests/controller/integration_test.go

```diff
@@ -249,6 +249,10 @@ func Test_UpdateNATSCR(t *testing.T) {
 func Test_DeleteNATSCR(t *testing.T) {
 	t.Parallel()
 
+	if !*testEnvironment.EnvTestInstance.UseExistingCluster {
+		t.Skip("This test can only run with a real cluster as k8s garbage collector is used.")
```

Comment:
why should we simulate it if can run it simply with k3d? Apart from this, the deletion logic belongs to the K8s. I wouldn't extra test it. 
We will anyway have e2e tests that cover this e2e.

Reply:
1. We should not be depending on garbage collection, if we need to delete something, we should explicitly delete in our codebase.
2. It would be better if we have one environment where all the tests are working. Either envtest or k3d. But not like half of the tests works with entest and half with k3d. We can easily run this test with envtest with manual simlulation of garbage collection. In this way, this test will work on both envtest and k3d.

Lets discuss in tech talk if you have more points?

Reply:
I absolutely would not implement it ourselves if K8s can do it. If K8s does it, it is stable, no effort from our side (logic, tests code).
Yes, let's discuss with the team.

Reply:
Unfortunately, PVC and garbage collection doesn't work with local envtest. I left only PVCs as we agreed, but it is not created in the local envtest, but works with a existing (k3d, gardener) clusters.